apps/backend/src/index.ts

```diff
@@ -31,7 +33,11 @@ app.use("/", healthRoutes);
 
 app.locals.intersectionState = {} as IntersectionState;
 
+const server = http.createServer(app);
 const PORT = process.env.PORT || 8080;
-app.listen(PORT, () => {
+server.listen(PORT, () => {
   console.log(`Server is running on port ${PORT}`);
-});
\ No newline at end of file
+});
+
+// Use same server for websocket connection
+export const wsServer = new WebSocketServer({ server })
```

Comment:
I thought the vercel issue may relate to http and ws not sharing the same port. This change allows ws to piggyback on the same server as http. Alas, it did not solve the vercel issue :-(

Reply:
@AndrewCLu -- another hypothesis is that the backend server hasn't been deployed in render, so it's missing the websocket server.

Reply:
I'm pretty sure this is the issue, resolving.